sklearn/preprocessing/data.py

```diff
@@ -323,7 +323,7 @@ def transform(self, X):
         """
         check_is_fitted(self, 'scale_')
 
-        X = check_array(X, copy=self.copy, ensure_2d=False)
+        X = check_array(X, copy=self.copy, ensure_2d=False, dtype=FLOAT_DTYPES)
```

Comment:
:confused: is that not going to cast a int32 dtype into a float64:

```python
import numpy as np

from sklearn.utils.validation import check_array, FLOAT_DTYPES
check_array(np.arange(10, dtype='i4'), dtype=FLOAT_DTYPES).dtype
```

returns dtype('float64').

On the other hand as_float_array seems to do the right thing.